java-checks/src/main/java/org/sonar/java/checks/UselessImportCheck.java

```diff
@@ -52,6 +54,8 @@
 @Rule(key = "S1128")
 public class UselessImportCheck extends BaseTreeVisitor implements JavaFileScanner {
 
+  private static final Pattern NON_WORDS_OR_WHITE_SPACES_CHARACTERS = Pattern.compile("(?:\\W|\\s)+");
```

Comment:
The `\\s` is redundant here. `\\W` already matches everything matched by `\\s`. So this can just be `"\\W+"`